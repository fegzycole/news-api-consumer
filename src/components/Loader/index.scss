@import "../../styles/palette";
@import "../../styles/mixins";

.Loader_Container {
  width: 100%;
  height: 600px;
  @include flex-styles(column, center, center);
}

.Loader {
  display: flex;

  .Circle {
    position: relative;
    width: 0.8em;
    height: 0.8em;
    margin: 0.2em;
    border-radius: 50%;

    &::before {
      position: absolute;
      content: "";
      width: 100%;
      height: 100%;
      background: inherit;
      border-radius: inherit;
      animation: wave 2s ease-out infinite;
    }

    @for $i from 1 through 5 {
      &:nth-child(#{$i}) {
        background: nth($colors, $i);

        &::before {
          animation-delay: $i * 0.2s;
        }
      }
    }
  }
}

@keyframes wave {
  50%,
  75% {
    transform: scale(2.5);
  }

  80%,
  100% {
    opacity: 0;
  }
}
