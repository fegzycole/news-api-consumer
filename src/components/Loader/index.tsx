import "./index.scss";

const Loader = () => {
  return (
    <div className="Loader_Container">
      <div className="Loader">
        <div className="Circle"></div>
        <div className="Circle"></div>
        <div className="Circle"></div>
        <div className="Circle"></div>
        <div className="Circle"></div>
      </div>
    </div>
  );
};

export default Loader;
