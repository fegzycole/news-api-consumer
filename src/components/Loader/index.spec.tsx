import { render } from "@testing-library/react";

import Loader from "./index";

describe("<Loader />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match the Loader component snapshot", () => {
    const view = render(<Loader />);
    expect(view.container).toMatchSnapshot();
  });
});
