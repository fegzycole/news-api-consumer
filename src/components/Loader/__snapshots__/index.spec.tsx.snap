// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match the Loader component snapshot 1`] = `
<div>
  <div
    class="Loader_Container"
  >
    <div
      class="Loader"
    >
      <div
        class="Circle"
      />
      <div
        class="Circle"
      />
      <div
        class="Circle"
      />
      <div
        class="Circle"
      />
      <div
        class="Circle"
      />
    </div>
  </div>
</div>
`;
